app.js

```diff
@@ -27,7 +27,12 @@ const pipeline = util.promisify(stream.pipeline);
 const maxFields = 1000;
 const debug = false;
 
-const isDirectory = async (filePath) => (await fs.lstat(filePath)).isDirectory();
+const isDirectory = async (filePath) => {
+  return (
+    (await fs.stat(filePath)).isDirectory() ||
```

Comment:
do we need to check both? does'nt lstat check the target if it's a symbolic link?

> The lstat() function shall be equivalent to [stat()](https://pubs.opengroup.org/onlinepubs/000095399/functions/stat.html), except when path refers to a symbolic link. In that case lstat() shall return information about the link, while [stat()](https://pubs.opengroup.org/onlinepubs/000095399/functions/stat.html) shall return information about the file the link references.

https://pubs.opengroup.org/onlinepubs/000095399/functions/lstat.html

Reply:
actually, shouldn't we only check for `fs.stat`? because we're interested in the target of the link?

Reply:
Yes.. But you still need lstat for a symlink to a directory to show up as directory and not file on other OS. For some reason, `fs.lstat` doesn't work on the `~/storage/shared` symlink in Termux.. Even though it is a symlink to the `/storage/emulated/0` directory on Android, NodeJS `fs.lstat` identifies it as a file but `fs.stat` identifies it as a directory..

![IMG_20240606_173749](https://github.com/mifi/ezshare/assets/90985774/994261fd-222f-46d4-b916-c2561f5b5cbd)